### PR TITLE
chore: don't detach CdpSession

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Allow the Mapper to handle it's own clean-up, else w get errors in the logs + we send too many commands.
Users should clean the CdpSession on their own if created via `createCDPSession`

### CODE REVIEW

**Comment by OrKoN** [comment]
Does it solve any bug/issue?

**Comment by Lightning00Blade** [comment]
There were Errors in the Logs, they are always cleaned up before the make the call so we get errors instead.

File: packages/puppeteer-core/src/bidi/CDPSession.ts [hunk 1]
```diff
@@ -86,10 +88,13 @@ export class BidiCdpSession extends CDPSession {
...
-        sessionId: this.id(),
-      });
+      if (this.#userCreated) {
+        await this.frame.client.send('Target.detachFromTarget', {
+          sessionId: this.id(),
+        });
+      }
     } finally {
       BidiCdpSession.sessions.delete(this.id());
       this.#detached = true;
```
Comment (OrKoN) [inline packages/puppeteer-core/src/bidi/CDPSession.ts]: can we learn the mapper closed the session and set the detached flag? I would rather suppress the error than introduce the notion of the user created sessions

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/bidi/CDPSession.ts]: We can just not call it. As the this is done on `browsingContext.constextDestroyed` event, and let the user handle the CDP session as they please.

Comment (OrKoN) [reply packages/puppeteer-core/src/bidi/CDPSession.ts]: We probably only need the BidiCdpSession.sessions.delete bit then

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/bidi/CDPSession.ts]: OK, I will update the PR and title

File: packages/puppeteer-core/src/bidi/Frame.ts [hunk 2]
```diff
@@ -106,7 +106,7 @@ export class BidiFrame extends Frame {
     this.browsingContext.on('closed', () => {
       for (const session of BidiCdpSession.sessions.values()) {
         if (session.frame === this) {
-          void session.detach().catch(debugError);
+          session.onClose();
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/bidi/Frame.ts:109]: If the `BrowserContext` is closed Mapper the session is already closed so we need to only remove it from the map.

**Review by OrKoN — APPROVED** [review]
